src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -367,23 +368,32 @@ object Task {
     def apply(mesosTaskId: MesosProtos.TaskID): Id = apply(mesosTaskId.getValue)
 
     /**
-      * Create a taskId for a pod instance's task. This will create a taskId designating the instance and each
-      * task container's name. It may be used for reservations for persitent pods as well.
+      * Create a taskId for an app or pod instance's reservation. This will create a taskId designating the instance and
+      * each task container's name.
       *
       * @param instanceId the ID of the instance that this task is contained in
       * @param container the name of the task as per the pod container config.
       */
-    def forInstanceId(instanceId: Instance.Id, container: Option[MesosContainer] = None): Id = EphemeralOrReservedTaskId(instanceId, container.map(_.name))
+    def forReservation(instanceId: Instance.Id, container: Option[MesosContainer] = None): Id = EphemeralOrReservedTaskId(instanceId, container.map(_.name))
```

Comment:
The method name says `forReservation` however we return `EphemeralOrReservedTaskId`. Do you think we should make it more concise?

Reply:
Unfortunately we cannot name the type just `ReservedTaskId` since old ephemeral task uses the same id pattern. We could rename `forReservation` but it should only be used for reservations so that does not make much sense to me either. I'm happy for any suggestion.

Reply:
Is there a reason why we need to have reserved / non-reserved semantics in our task ids? Ideally, I think we'd just have task-ids, since both have incarnations.

Reply:
I ➕ @meln1k confusion, I don't get this part as well :( Maybe we can clean this up a bit more... We can chat tomorrow

Reply:
Ok, I understand the confusion. The issue is the ids for reservations are the same as the *old* ephemeral task ids. When the ids string of a reservation or a task is parsed there is no way to tell whether it should be `ReservationId` or `EphemerlId`. Hence `EphemeralOrReservationId`. Unfortunately this means we have to use the later when we create reservation ids. I can try a follow up and really separate them but this is already much cleaner than on master since we at least call `forReservation` in the right places. This will make a follow up easier. If you want to know what I went through take a look at #6076.

That aside I'm surprised you all are surprised 🙂. Reservation ids and ephemeral ids will be the same until #6677 lands. With #6677 we simply get a new type for ephemeral task ids. Also, keep in mind reservation ids are *not* resident task ids! Ideally reservation ids would be complete separate but so far we cannot do that.